src/components/grid/index.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { FC, Children, useCallback, useEffect } from 'react'
+import useMousetrap from 'react-hook-mousetrap'
+import { useDispatch, useSelector } from 'react-redux'
+import { AnyAction, Dispatch } from 'redux'
+
+import { createGrid, IReducer, selectBlock, fillBlock } from 'reducers'
+import { BLOCK_COORDS, GRID, INDEX, N, NUMBERS } from 'typings'
+
+import Block from './block'
+import { Container, Row } from './styles'
+
+interface IState {
+  selectedBlock?: BLOCK_COORDS
+  selectedValue: N
+  solvedGrid?: GRID
+}
+
+const Grid: FC = () => {
+  const state = useSelector<IReducer, IState>(
+    ({ selectedBlock, workingGrid, solvedGrid}) => ({
+      selectedBlock,
+      selectedValue:
+        workingGrid && selectedBlock
+          ? workingGrid[selectedBlock[0]][selectedBlock[1]]
+          : 0,
+          solvedGrid,
+    })
+  )
+  const dispatch = useDispatch<Dispatch<AnyAction>>()
+
+  const create = useCallback(() => dispatch(createGrid()), [dispatch])
+
+  const fill = useCallback(
+    (n: NUMBERS) => {
+      if (state.selectedBlock && state.selectedValue === 0)
+        dispatch(fillBlock(n, state.selectedBlock))
+    },
+    [dispatch, state.selectedBlock, state.selectedValue]
+  )
+
+  function moveDown() {
+    if (state.selectedBlock && state.selectedBlock[0] < 8)
+      dispatch(
+        selectBlock([
+          (state.selectedBlock[0] + 1) as INDEX,
+          state.selectedBlock[1],
+        ])
+      )

```

Comment:
I understand the tutorial may be showing if conditions like this, but it is a VERY BAD practice to do this. Having an if without key blocks to contain it is highly error prone and can be very confusing to other developers when they read your code. I encourage you to instead of doing if conditions where the condition is simply indented in the next line, to actually use proper if blocks:

```
if (condition) {
 ... do your stuff here
}
```

This is much more clear and much easier to follow and read when you share the code with other devs. It would be good to make this change across all your files wherever you're doing any if conditions. This particular example would change to:

```
if (state.selectedBlock && state.selectedBlock[0] < 8) {
      dispatch(
        selectBlock([
          (state.selectedBlock[0] + 1) as INDEX,
          state.selectedBlock[1],
        ])
      )
}
```

Reply:
OMG coz I thought this way  if () without {}  using with .tsx

Reply:
haha nono, it's normal ifs, it's likely the video instructor was showing it for "convenience" but that convenience can bite us back very hard if we miss the indentation and other things